test/e2e/components_upgrade_test.go

```diff
@@ -0,0 +1,71 @@
+// +build e2e
+
+package e2e
+
+import (
+	"path/filepath"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/semver"
+	releasev1alpha1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+	"github.com/aws/eks-anywhere/test/framework"
+)
+
+const (
+	prodReleasesManifest = "https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml"
+	latestReleasePath    = "bin/latest-release"
+	releaseBinaryName    = "eksctl-anywhere"
+)
+
+func getLatestMinorReleaseFromMain(test *framework.ClusterE2ETest) (binaryPath string) {
+	reader := cluster.NewManifestReader()
+	test.T.Logf("Reading prod release manifest %s", prodReleasesManifest)
+	releases, err := reader.GetReleases(prodReleasesManifest)
+	if err != nil {
+		test.T.Fatal(err)
+	}
+
+	var latestRelease *releasev1alpha1.EksARelease
+	for _, release := range releases.Spec.Releases {
+		if release.Version == releases.Spec.LatestVersion {
+			latestRelease = &release
+			break
+		}
+	}
+
+	if latestRelease == nil {
+		test.T.Fatalf("Releases manifest doesn't contain latest release %s", releases.Spec.LatestVersion)
+	}
+
+	latestReleaseBinaryFolder := filepath.Join(latestReleasePath, latestRelease.Version)
+	latestReleaseBinaryPath := filepath.Join(latestReleaseBinaryFolder, releaseBinaryName)
+
+	return latestReleaseBinaryPath
+}
+
+func getLatestMinorReleaseFromReleaseBranch(test *framework.ClusterE2ETest, releaseBranchVersion *semver.Version) (binaryPath string) {
+	reader := cluster.NewManifestReader()
+	test.T.Logf("Reading prod release manifest %s", prodReleasesManifest)
+	releases, err := reader.GetReleases(prodReleasesManifest)
+	if err != nil {
+		test.T.Fatal(err)
+	}
+
+	var latestPrevMinorRelease *releasev1alpha1.EksARelease
+
+	for _, release := range releases.Spec.Releases {
+		releaseVersion, err := semver.New(release.Version)
+		if releaseVersion.IsPrevMinorVersion(releaseBranchVersion) && releaseVersion.GreaterThan(latestPrevMinorRelease.Version) {
```

Comment:
I'm not sure how this works
* `latestPrevMinorRelease` is of type `*releasev1alpha1.EksARelease`, how can it be compared to `releaseVersion`? Also, it's nil at the start
* I'm not sure the `IsPrevMinorVersion`part will work for all the scenarios. What if they belong to different mayor versions? This might be enough for now, but a more robust way to do it would be `releaseVersion < releaseBranchVersion && releaseVersion.Minor != releaseBranchVersion.Minor && releaseVersion > latestPrevMinorRelease` (pseudo code)